src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -302,9 +303,20 @@ export class ConnectionDialogService implements IConnectionDialogService {
 
 	private handleShowUiComponent(input: OnShowUIResponse) {
 		if (input.selectedProviderType) {
-			this._currentProviderType = input.selectedProviderType;
+			// If the call is for specific providers
+			if (this._params && this._params.providers) {
+				this._params.providers.forEach((provider) => {
+					if (input.selectedProviderType === this._providerNameToDisplayNameMap[provider]) {
+						this._currentProviderType = provider;
+					}
```

Comment:
So we're fine with this being unchanged if the provider type doesn't exist in _params.providers? What does that even mean? 

Also shouldn't you be doing something (clearing the value?) if the name doesn't exist in the params?